_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -0,0 +1,356 @@
+	<tr>
+		<th id="col1">Column 1</th>
+		<th id="col2">Column 2</th>
+	</tr>
+	<tr>
+		<td></td>
+	</tr>
+	<tr>
+		<td headers="col2"></td>
+	</tr>
+</table>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The column header ("Column 2"), does not have an assigned cell within the same `table` element.
+
+```html
+<table>
+	<thead>
+		<tr>
+			<th>Column 1</th>
+			<th>Column 2</th>
+		</tr>
+	</thead>
+	<tbody>
+		<tr>
+			<td>15%</td>
+		</tr>
+	</tbody>
+</table>
+```
+
+#### Failed Example 2
+
+The column header ('Column2') does not have an assigned cell within the same `table` element. In this case the usage of `headers` attribute removes cell association to the column.
+
+```html
+<table>
+	<tr>
+		<th id="col1">Column 1</th>
+		<th id="Column2">Column 2</th>
+	</tr>
+	<tr>
+		<td></td>
+		<td headers="col1"></td>
+	</tr>
+</table>
+```
+
+#### Failed Example 3
+
+The column header ('Col B') does not have an assigned cell within the same `table` element.
+
+```html
+<div role="grid">
+	<div role="row">
+		<div role="columnheader">Col A</div>
+		<div role="columnheader">Col B</div>
+	</div>
+	<div role="row">
+		<div role="gridcell">1A</div>
+		<div role="gridcell">1B</div>
+	</div>
+	<div role="row">
+		<div role="gridcell">2A</div>
+	</div>
+</div>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+There is no element with a [semantic role][] of header within the table.
+
+```html
+<table>
+	<tr>
+		<td>12:00</td>
+	</tr>
+</table>
+```
+
+#### Inapplicable Example 2
+
+There are no element with a [semantic role][] of header.
+
+```html
+<table></table>
+```
+
+#### Inapplicable Example 3
+
+There are no elements with a [role][semantic role] of header within the table.
+
+```html
+<table>
+	<tr>
+		<th role="cell">Column A</th>
+	</tr>
+	<tr>
+		<td>Cell A</td>
+	</tr>
+</table>
+```
+
+#### Inapplicable Example 4
+
+The only element with a [role][semantic role] of header is not [visible][].
+
+```html
+<table>
+	<tr>
+		<th style="display: none;">Column A</th>
+	</tr>
+	<tr>
+		<td>Cell A</td>
+	</tr>
+</table>
+```
+
+#### Inapplicable Example 5
+
+The only element with a [role][semantic role] of header is not [included in the accessibility tree][].
+
+```html
+<table>
+	<tr>
+		<th aria-hidden="true">Column A</th>
+	</tr>
+	<tr>
+		<td>Cell A</td>
+	</tr>
+</table>
+```
+
+#### Inapplicable Example 6
+
+The only element with a [role][semantic role] of header does not have a closest ancestor in the [flat tree][] with a [semantic role][] of either [table][] or [grid][]
+
+```html
+<div>
+	<tr>
+		<th>Column A</th>
+	</tr>
+</div>
+```
+
+#### Inapplicable Example 7
+
+The only element with header [role][semantic role] is part of a table which is not [visible][].
+
+```html
+<html>
+	<style>
+		.notInPage {
+			position: absolute;
+			left: -9999px;
+			top: -9999px;
+		}
+	</style>
+	<body>
+		<table class="notInPage">
+			<tr>
+				<th>Time</th>
+			</tr>
+			<tr>
+				<td>24:00</td>
+			</tr>
+		</table>
+	</body>
+</html>
+```
+
```

Comment:
This comment relates to my comment on the rule Assumptions about visibility: Though this table is not visible on the page, it is still in the accessibility tree and can be navigated with AT.